components/core/src/clp/BoundedReader.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef CLP_BOUNDEDREADER_HPP
+#define CLP_BOUNDEDREADER_HPP
+
+#include "ReaderInterface.hpp"
+
+namespace clp {
+/**
+ * BoundedReader is a ReaderInterface designed to wrap other ReaderInterfaces and prevent users
+ * from reading or seeking beyond a certain point in the underlying input stream.
+ *
+ * This is useful when the underlying input stream is divided into several logical segments and we
+ * want to prevent a reader for an earlier segment consuming any bytes from a later segment. In
+ * particular, reading part of a later segment may force the reader for that later segment to seek
+ * backwards, which can be either inefficient or impossible for certain kinds of input streams.
+ */
+class BoundedReader : public ReaderInterface {
+public:
+    // Constructor
+    explicit BoundedReader(ReaderInterface* reader, size_t bound)
+            : m_reader(reader),
+              m_bound(bound) {
```

Comment:
```suggestion
            : m_reader{reader},
              m_bound{bound} {
```
We should enforce `{}` initializations in constructors' initialization list